src/Sulu/Bundle/ContentBundle/Document/Subscriber/PublishSubscriber.php

```diff
@@ -111,6 +119,11 @@ public function createNodeInPublicWorkspace(PersistEvent $event)
      */
     public function removeNodeFromPublicWorkspace(RemoveEvent $event)
     {
+        $metadata = $this->metadataFactory->getMetadataForClass(get_class($event->getDocument()));
```

Comment:
Extract `$event->getDocument()` into own variable, since it is used again afterwards.